### PR TITLE
Phase-2 Inner Tracker: combined backport of 33495 and 33508

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a combined backport of the commit of PRs #33495 and #33508. 
The changes propose here are needed to fix some minor issues with the determination of the template CPE reconstruction in the case of the Pixel 3D sensors for the Phase-2 inner tracker.
More discussion is available at: https://github.com/cms-sw/cmssw/pull/33495#issuecomment-824616464.
Commit https://github.com/cms-sw/cmssw/pull/33792/commits/2044fd55de151707a1bfbf508e76345d16508ce4 removes the changes that break the current online menus in 11.3.X (introduced after https://github.com/cms-sw/cmssw/pull/33792#discussion_r635894241).
A backport has become necessary in order to allow the production of a small set of samples for detailed POG-level checks on different phase-2 Inner Tracker sensor choices to be handled in CMSSW_11_3_X.

#### PR validation:

Tested and validated privately in the master branch. It compiles in the 11.3.X branch. The branch has also been tested successfully with `addOnTests.py`.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

This is a backport of the commits of PRs  #33495 and #33508. 

@cms-sw/upgrade-l2 @emiglior FYI.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -129,11 +129,26 @@ def customiseFor2018Input(process):
 
     return process
 
+def customiseFor33495(process):
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:132]: Should be customiseFor33792.

However, could this customisation and changes in this file just be removed, and the parameters in question be labelled "obsolete" in fillDescriptions of the plugins, such that configs containing them would still work?
Reason is that online does not use this customisation, and we also do not want to break old 11_3 menus.

Comment (mmusich) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: @Martin-Grunewald, commit https://github.com/cms-sw/cmssw/pull/33792/commits/2044fd55de151707a1bfbf508e76345d16508ce4 addresses your comment and incidentally remove the HLT pending flag.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by slava77** [comment]
> Reco comparison results: 3938 differences found in the comparisons

this is a closed release and differences appear in all eras (not restricted to phase-2).
Something should be done: either modify to have no changes or see if/how an exception can be made.

**Comment by mmusich** [comment]
> this is a closed release and differences appear in all eras (not restricted to phase-2).

Before doing anything, I'd like to understand if the existing releases in this cycle have been used or are expected to be used for anything other than MWGR.

**Comment by slava77** [comment]
> > this is a closed release and differences appear in all eras (not restricted to phase-2).
> 
> Before doing anything, I'd like to understand if the existing releases in this cycle have been used or are expected to be used for anything other than MWGR.

was this discussed already outside of the github thread?
@silviodonato @qliphy 
please advise on ability to make changes in 11_3_X.
Thank you.

**Comment by qliphy** [comment]
@slava77 @mmusich    There will be a production in 11_3_X for the Run-3 preparation, but it has not started yet.   (@ may comment on this.)
BTW:   @silviodonato and me think the differences here should come from the backport related to https://github.com/cms-sw/cmssw/pull/33495, which is basically a bugfix. The differences seem also to be in line with those in master.

**Comment by mmusich** [comment]
type bug-fix

**Comment by srimanob** [comment]
Following the PPD news last week (https://indico.cern.ch/event/1034467/contributions/4344266/attachments/2248271/3813570/21-05-20_News_PPD.pdf, slide 5), there is no plan for production with 11_3. The goals of 11_3 include Geant4 10.7 and GPU. If we need GPU production for validation, it should be OK too because changes should appear in both CPU-only and CPU-GPU. 

This PR (together with https://github.com/cms-sw/cmssw/pull/33795 from MTD) are needed for small Phase-2 high-stat production. There is a technical reason that it can't be done as relvals in 12_0. I will mention this at the ORP tomorrow.

**Comment by srimanob** [comment]
+Upgrade

@cms-sw/reconstruction-l2 Do you agree for this backport which changes reco in 11_3? Thanks.

File: RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc [hunk 2]
```diff
@@ -89,8 +89,8 @@ void PixelCPEFastESProducer::fillDescriptions(edm::ConfigurationDescriptions& de
   // specific to PixelCPEFastESProducer
   desc.add<std::string>("ComponentName", "PixelCPEFast");
   desc.add<edm::ESInputTag>("MagneticFieldRecord", edm::ESInputTag());
-  desc.add<bool>("useLAAlignmentOffsets", false);
-  desc.add<bool>("DoLorentz", false);
+  desc.add<bool>("useLAAlignmentOffsets", false)->setComment("deprecated");
+  desc.add<bool>("DoLorentz", false)->setComment("deprecated");
```
Comment (slava77) [inline RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc:93]: shouldn't this be `addOptional`?
I was looking at https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Optional_Parameters
by the context I would think that addObsolete would be a more proper naming, but it seems like it's used for other purposes.
The context here is related to https://github.com/cms-sw/cmssw/pull/33792#discussion_r635894241 `the parameters in question be labelled "obsolete" in fillDescriptions of the plugins, such that configs containing them would still work`

@makortel @Dr15Jones 
please check/advise.
Thank you.

Comment (makortel) [reply RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc:93]: `addOptional()` without a default value is a closest feature for declaring obsolete parameters in the `fillDescriptions()`, i.e. such parameter will not be part of the generated cfi file, configuration validation does nothing such parameter is missing, and defining such parameter does not cause error.

Comment (mmusich) [reply RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc]: @slava77 addressed at https://github.com/cms-sw/cmssw/pull/33792/commits/64ab2c307e4be193ada9aa6eb5b51252fd2893da

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33792	64ab2c3

- code changes are in line with the PR description and the follow up review
    - the backport from #33508 is exact; the backport from #33495 has a technical difference to keep consistent support of the old HLT configurations 
    - physics changes in tracking affect all eras, but can be interpreted as a bugfix (although the fix is for a part of the code that was present for a few release cycles and would nominally be described as a feature update)
- jenkins tests pass and comparisons with the baseline show small differences which start in tracks and propagate downstream to dependent values
    - the last test of this PR also includes a large set of differences in wf 11634.911, which is known to be non-repeatable at the SIM step and can be ignored

**Comment by srimanob** [comment]
+Upgrade

(re-sign)

**Comment by srimanob** [comment]
@cms-sw/alca-l2 @cms-sw/pdmv-l2 Could you please and comment? We expect this to be in the next 11_3 release by early next week. Thanks.

**Comment by tlampen** [comment]
+alca

Backport of #33508 (see comments therein) and #33495 (bugfix), both fully signed.

**Comment by qliphy** [comment]
@cms-sw/pdmv-l2  Kindly ping.  The next 11_3 release will be made by early next week, and this PR is needed.

**Comment by qliphy** [comment]
urgent

**Comment by chayanit** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**